src/api/v4/error-handling/carto-error-extender.js

```diff
@@ -0,0 +1,82 @@
+var _ = require('underscore');
+var ERROR_LIST = require('./error-list');
+
+/**
+ * Get two parameters to enrich a CartoError.
+ * - friendlyMessage: A easy to understand error description.
+ * - errorCode: A unique errror code
+ * 
+ * @param {CartoError} cartoError
+ * 
+ * @returns {object} - An object containing a friendly message and a errorCode
+ */
+function getExtraFields (cartoError) {
+  var errorlist = _getErrorList(cartoError);
+  var listedError = _getListedError(cartoError, errorlist);
+
+  return {
+    friendlyMessage: listedError.friendlyMessage,
+    errorCode: listedError.errorCode
+  };
+}
+
+/**
+ * 
+ * @param {CartoError} cartoError 
+ */
+function _getErrorList (cartoError) {
+  return ERROR_LIST[cartoError.origin] && ERROR_LIST[cartoError.origin][cartoError.type];
+}
+
+/**
+ * Get the listed error from a cartoError, if no listedError is found return a generic
+ * unknown error.
+ * @param {CartoError} cartoError 
+ */
+function _getListedError (cartoError, errorList) {
+  var errorListkeys = _.keys(errorList);
+  var key;
+  for (var i = 0; i < errorListkeys.length; i++) {
+    key = errorListkeys[i];
+    if (errorList[key].messageRegex.test(cartoError.message)) {
+      return {
+        friendlyMessage: _replaceRegex(cartoError, errorList[key]),
+        errorCode: _buildErrorCode(cartoError, key)
+      };
+    }
+  }
+
+  // When cartoError not found return generic values
+  return {
+    friendlyMessage: cartoError.message,
+    errorCode: _buildErrorCode(cartoError, 'unknown-error')
+  };
+}
+
+/**
+ * Replace $0 with the proper paramter in the listedError regex to build a friendly message
+ */
+function _replaceRegex (cartoError, listedError) {
+  var match = cartoError.message && cartoError.message.match(listedError.messageRegex);
+  if (match && match.length > 1) {
+    return listedError.friendlyMessage.replace('$0', match[1]);
```

Comment:
The argument `cartoError` can have an undefined `.friendlyMessage` property, throwing an error in this case.